File: libs/language/src/lib/mapping/__tests__/schema/personObjectExample.ts
import z from 'zod';
import {
  blueIdField,
  blueNodeField,
  withTypeBlueId,
} from '../../../../schema/annotations';
import { personSchema } from './person';

export const personObjectExampleSchema = withTypeBlueId(
  'PersonObjectExample-BlueId'
)(
  z.object({
    alice1: blueIdField(),
    alice2: blueNodeField(),
    alice3: z.record(z.string(), z.unknown()).optional(),
    alice4: personSchema.optional(),
    alice5: personSchema.optional(),
  })
);

File: references/java-sdk/src/test/java/blue/language/mapping/model/PersonObjectExample.java
package blue.language.mapping.model;

import blue.language.model.BlueId;
import blue.language.model.Node;
import blue.language.model.TypeBlueId;

import java.util.Map;

@TypeBlueId("PersonObject-BlueId")
public class PersonObjectExample {
    @BlueId
    public String alice1;
    public Node alice2;
    public Map<String, Object> alice3;
    public Person alice4;
    public Person alice5;
}
